tests/test_general.py

```diff
@@ -267,7 +265,23 @@ def test_model_callable():
     for arg_name, name in zip(('x', 'y', 'a', 'b'), inspect_sig.signature(model).parameters):
         assert arg_name == name
 
-    # From Model __init__ directly
+
+@pytest.mark.skip(reason=("Test fails a proportion of the time because `z_1`, "
+                          "`z_2` and `z_3` are not necessarily ordered as "
+                          "expected so the assert statements fail. `z_1` is "
+                          "frequenty equal to either 36 or 72."))
```

Comment:
My suggestion for fixing this is to sort the values:
`z_1, z_2, z_3 = sorted(model(3, 3, 2, 2))`

Reply:
@pckroon, thanks! Yup, the reason I marked it for skipping was because I didn't know how to fix it. Will indeed follow your suggestion and double check it fixes it, then unmark the test for skipping.

Reply:
Alternatively we could consider this a bug. It's a little bit silly that when you create a model with a list you don't get the resulting values in the same order, even if it's not the recommended way of creating a model. 
@tBuLi Opinion?

Reply:
FYI simply applying `sorted()` to it as you suggested doesn't fix it. I'm looking in to a bug fix for it now. Will keep you posted.

Reply:
It seems the issue is coming from `BaseModel._init_from_dict` which toposorts the values in the model and then stores this "sorted" ordering in its internal state. In the case of this test there is no preferred ordering of the three expressions in the model as each only depends on `a`, `b` and `x`, all of which are independent of each other.

I agree it makes sense to have the resulting values returned in the same order in which they are supplied. One workaround would be to internally store in the `BaseModel` the order of the model's expressions when supplied as then return the values according to this order? Would you like me to try implement this as part of this PR, or should I leave this test marked with a skip, file an issue relating to this bug and address in a separate PR?

Reply:
If you're willing to take a stab at fixing the underlying issue that should be a separate PR (makes review easier). 
I think BaseModel is the right place to "remember" the order of the components. Note that the order in which we should return them should be completely independent of what the topological order is.

Reply:
I have opened an issue for this in #318. With this issue and the above do we agree that it is okay to leave this specific test marked as for skipping and resolve this review comment?